libs/market-depth/src/lib/orderbook.spec.tsx

```diff
@@ -1,10 +1,165 @@
-import { render } from '@testing-library/react';
-
-import Orderbook from './orderbook';
+import { render, fireEvent, waitFor, screen } from '@testing-library/react';
+import { generateMockData } from './orderbook-data';
+import { Orderbook, rowHeight } from './orderbook';
 
 describe('Orderbook', () => {
-  it('should render successfully', () => {
-    const { baseElement } = render(<Orderbook data={null} decimalPlaces={4} />);
-    expect(baseElement).toBeTruthy();
+  const params = {
+    numberOfSellRows: 100,
+    numberOfBuyRows: 100,
+    midPrice: 122900,
+    bestStaticBidPrice: 122905,
+    bestStaticOfferPrice: 122895,
+    decimalPlaces: 3,
+    overlap: 10,
+    indicativePrice: 122900,
+    indicativeVolume: 11,
+    resolution: 1,
+  };
+  const onResolutionChange = jest.fn();
+  const decimalPlaces = 3;
+  it('should scroll to mid price on init', async () => {
+    window.innerHeight = 11 * rowHeight;
+    const result = render(
+      <Orderbook
+        decimalPlaces={decimalPlaces}
+        {...generateMockData(params)}
+        onResolutionChange={onResolutionChange}
+      />
+    );
+    await waitFor(() => screen.getByTestId('bid-vol-122900'));
```

Comment:
Chore (blocking): used in lots of places:

```suggestion
    await waitFor(() => screen.getByTestId(`bid-vol-${params.midPrice}`));
```